src/app/core/notifications/notifications_manager.nim

```diff
@@ -0,0 +1,170 @@
+    let details = NotificationDetails(notificationType: NotificationType.NewContactRequest, sectionId: sectionId)
+    self.processNotification(title, message, details)
+
+  proc onNewCommunityMembershipRequestNotification*(self: NotificationsManager, title: string, message: string, 
+    sectionId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.JoinCommunityRequest, sectionId: sectionId)
+    self.processNotification(title, message, details)
+
+  proc onMyRequestToJoinCommunityHasBeenAcccepted*(self: NotificationsManager, title: string, message: string, 
+    sectionId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.MyRequestToJoinCommunityAccepted, 
+    sectionId: sectionId)
+    self.processNotification(title, message, details)
+
+  proc onMyRequestToJoinCommunityHasBeenRejected*(self: NotificationsManager, title: string, message: string, 
+    sectionId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.MyRequestToJoinCommunityRejected, 
+    sectionId: sectionId)
+    self.processNotification(title, message, details)
+
+  proc processNotification(self: NotificationsManager, title: string, message: string, details: NotificationDetails) =
+    ## This is the main method which need to be called to process an event according to the preferences set in the 
+    ## notifications panel of the settings section.
+    ## 
+    ## This method determines whether a notification need to be displayed or not, what notification to display, whether
+    ## to display App or OS notification, what level of anonymous to apply, whether to play sound or not and so...
+
+    var finalTitle = title
+    var finalMessage = message
+
+    # Check if notification need to be displayed
+    if(singletonInstance.localAccountSensitiveSettings.getNotificationSetting() == NOTIFY_NOTHING_ABOUT and
+      (details.notificationType == NotificationType.NewMessage or 
+      details.notificationType == NotificationType.NewMention)):
+      return
+
+    if(singletonInstance.localAccountSensitiveSettings.getNotificationSetting() == NOTIFY_JUST_ABOUT_MENTIONS and
+      details.notificationType == NotificationType.NewMessage):
+      return
+
+    if(not singletonInstance.localAccountSensitiveSettings.getNotifyOnNewRequests() and
+      details.notificationType == NotificationType.NewContactRequest):
+      return
+
+    # Check anonymous level
+    if(singletonInstance.localAccountSensitiveSettings.getNotificationMessagePreviewSetting() == PREVIEW_ANONYMOUS):
+      finalTitle = "Status"
+      finalMessage = "You have a new message"
+    elif(singletonInstance.localAccountSensitiveSettings.getNotificationMessagePreviewSetting() == PREVIEW_NAME_ONLY):
+      finalMessage = "You have a new message"
+        
+    # Check whether to display APP or OS notification
+    if(singletonInstance.localAccountSensitiveSettings.getUseOSNotifications()):
```

Comment:
We only use OS notifications now. What we've seen with John is that we use OS notifications when the app is minimized, and we use "toast" notifications when the app is visible.

The old `useOSNotifications` setting can be deleted.

In summary, this condition can be changed to `if (app is minimized)` and you can delete the `useOSNotifications` setting and check box in the settings

Reply:
@jrainville let's keep it like it is now, cause this state maps what we have at this moment in Settings->Notifications tab, once we get new requirements we will updated this. But primarily is to have at least what we had.

Reply:
I wrote about this below, but we should really change it now. The OS notifications are not that useful in the state you made, because the app needs to be opened already and it's very annoying to receive OS notifications when you're already in the app.
Deleting the old code can be done later, but the check on app minimized would be good now.